### PR TITLE
📝  Logging documentation

### LINKED ISSUE #357: Describe logging feature in docs
## Description of issue / requirement to address

Bluemira implements some neat logging that allows information to be captured in files as well as on screen. However, this is implemented via handlers on global loggers, so anyone using that functionality will need to be conscious of the context in which they are modifying the loggers. This should hopefully avoid issues like those seen in #355.

## Proposed solution

Add clarity in docs about how the logging functionality works and how users should be aware of the global context that is adjusted by enabling the bluemira-style logging.

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Closes #357 

## Description

Documents our logging functionality in the developer section, with some examples.

I've also organised the developer section source files a bit

## Interface Changes

None

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: documentation/source/developer/logging.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                # Do stuff
+
+
+        # A cleaner but less customisable version
+        def my_function():
+            loglevel = get_log_level()
+            if loglevel == "DEBUG":
+                # Do stuff
+
+* For updated text (ex. progress bars) there are a few functions that do not print newlines
```
Comment (hsaunders1904) [inline documentation/source/developer/logging.rst:60]:
```suggestion
* For updated text (e.g. progress bars) there are a few functions that do not print newlines
```

File: documentation/source/developer/logging.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        from bluemira.base.look_and_feel import (
+            bluemira_debug_flush,
+            bluemira_print_flush,
+        )
+
+        bluemira_print_flush('my message')
+        bluemira_print_flush('my other message')  # overwrites the first message
+
+* If formatted output is not desired there are error and print functions the direct to the logger
```
Comment (hsaunders1904) [inline documentation/source/developer/logging.rst:73]:
```suggestion
* If formatted output is not desired, there are error and print functions that direct to the logger
```

File: documentation/source/developer/logging.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
+Logging in Bluemira
+-------------------
+
+To organise and have granularity in text (and potentially graphical) outputs
+we use some wrappers around python's built-in logging module.
+This allows us to both store the output in a log file but also be precise about
+how much information is presented at any one time.
```
Comment (hsaunders1904) [inline documentation/source/developer/logging.rst:7]:
```suggestion
This allows us to store the output in a log file whilst also being precise about
what information is displayed in standard out.
```

**Review by hsaunders1904 — APPROVED** [review]
Looks good! Only a couple of nitpicks

**Comment by hsaunders1904** [comment]
I'm happy for this to be merged. @CoronelBuendia you're also assigned, did you want to look at this as well before it's merged?

**Review by CoronelBuendia — APPROVED** [review]

**Pull request merged**